### ISSUE TITLE
XB271HUA - Not working

### ISSUE THREAD

**Issue by znbaboy** [xanderfrangos/twinkle-tray#461]
![image](https://user-images.githubusercontent.com/52028136/194772469-bed07cc1-b0dc-4ae5-ae00-aa955d87844d.png)

Works on my BenQ monitor but my Acer monitor doesn't respond to any brightness changes. 

Thanks

**Comment by alphataru** [comment]
Also have XB271HU, and it isn't working for me either. The slider works, but the brightness doesn't change.

**Comment by xanderfrangos** [comment]
Does modifying the Brightness value (VCP code 10) in ControlMyMonitor work for you? 
https://www.nirsoft.net/utils/control_my_monitor.html

**Comment by znbaboy** [comment]
> 

This doesn't work for me

**Comment by xanderfrangos** [comment]
Then it appears that the issue is that the monitor does not support the brightness being adjusted over DDC/CI. Unless a different VCP code controls the brightness for that monitor, there isn't anything I can do.

**Comment by znbaboy** [comment]
That's disappointing. The contrast works at least.

**Comment by xanderfrangos** [comment]
It's possible that a different VCP code actually adjusts the brightness for that monitor. You can try tinkering with ControlMyMonitor to see if any of them do it. If that's the case, there's a way to tell Twinkle Tray an alternate VCP code to use for the display.

Also it's worth mentioning that if the display does HDR and HDR is enabled, that Twinkle Tray probably won't work.

**Comment by znbaboy** [comment]
https://community.acer.com/en/discussion/576132/xb271hu-brightness-control-over-ddc-ci-protocol

Here they talk about being able to change brightness while running the monitor in "factory mode". 

I already tested ControlMyMonitor way back when you replied, just got to replying now. None of the other codes worked.

**Comment by xanderfrangos** [comment]
I'd say it's weird that my older XB270HU works just fine and the newer model doesn't, but this seems to happen all of the time with monitors. Sometimes manufacturers support DDC/CI and sometimes they don't.

**Comment by znbaboy** [comment]
Just tested it. If you enter Factory Mode and enable "Burn-in pattern" it seems to enable the code which allows brightness control. You can then close the menu and use the monitor like normal. The only issue is, if the monitor detects no signal (eg. your PC is in sleep/shutdown) It will play a burn-in pattern with flashing color so you will need to turn off your display physically,

I am going to test using a HDMI dummy to see if that prevents the monitor from playing the pattern. Although, even if it does, it will keep the display active in theory. If only Acer could have just programmed the firmware correctly smh

**Comment by znbaboy** [comment]
@xanderfrangos how is the power control being detected and available for some monitors? Im thinking if I can have the monitor auto powered off when my pc sleeps/shutsdown then that would be a decent work around

**Comment by just4747** [comment]
Has there been any fix for this since 2022 where it can work with the XB271HU without burn-in pattern turned on? It works for me with the setting on but having the monitor flash all colors every time it goes to sleep just sucks.

**Comment by znbaboy** [comment]
@just4747, no. Pretty sure its a bug in the monitors firmware. Even if they fixed it (which they wont), good luck updating the monitors firmware (there is no consumer accessible way to do so)

Sorry homie

**Comment by just4747** [comment]
> [@just4747](https://github.com/just4747), no. Pretty sure its a bug in the monitors firmware. Even if they fixed it (which they wont), good luck updating the monitors firmware (there is no consumer accessible way to do so)
> 
> Sorry homie

Damn that's too bad. Thanks.
